src/cclib/parser/daltonparser.py

```diff
@@ -752,20 +769,36 @@ def extract(self, inputfile, line):
         # @    Final gradient norm:           0.000003746706
         # ...
         #
+        if "Final HF energy" in line and not (hasattr(self, "mpenergies") or hasattr(self, "ccenergies")):
+            self.metadata["methods"] = "HF"
+        if "Final DFT energy" in line:
+            self.metadata["methods"] = "DFT"
+        if "This is a DFT calculation of type" in line:
+            self.metadata["functional"] = line.split()[-1]
+
         if "Final DFT energy" in line or "Final HF energy" in line:
             if not hasattr(self, "scfenergies"):
                 self.scfenergies = []
             temp = line.split()
             self.scfenergies.append(utils.convertor(float(temp[-1]), "hartree", "eV"))
 
         if "@   = MP2 second order energy" in line:
+            self.metadata["methods"] = "MP2"
             energ = utils.convertor(float(line.split()[-1]), 'hartree', 'eV')
             if not hasattr(self, "mpenergies"):
                 self.mpenergies = []
             self.mpenergies.append([])
             self.mpenergies[-1].append(energ)
 
+        if "Total CC2   energy:" in line:
```

Comment:
It needs to be changed to `Total CCSD`; CC2 is an approximation to CCSD and isn't the same thing.